compiler/src/dotty/tools/backend/jvm/BCodeSkelBuilder.scala

```diff
@@ -92,15 +106,28 @@ trait BCodeSkelBuilder extends BCodeHelpers {
 
       initJClass(cnode)
 
-      val hasStaticCtor = cd.symbol.methodSymbols exists (_.isStaticConstructor)
+      val methodSymbols = for (f <- cd.symbol.info.decls.toList if f.is(Flags.Method) && f.isTerm && !f.is(Flags.Module)) yield f
```

Comment:
Calling toList shouldn't be necessary to iterate over decls.

Reply:
Done